int32 userID
int32 timepoint
string message
